src/Surfnet/StepupMiddleware/ApiBundle/Controller/VerifiedSecondFactorController.php

```diff
@@ -65,10 +65,32 @@ public function getAction($id)
 
     public function collectionAction(Request $request)
     {
-        $this->denyAccessUnlessGranted(['ROLE_RA', 'ROLE_SS']);
+        $this->denyAccessUnlessGranted(['ROLE_RA']);
 
         $actorId = new IdentityId($request->get('actorId'));
 
+        $query = $this->buildVerifiedSecondFactorQueryFrom($request);
+        $query->authorizationContext = $this->institutionAuthorizationService->buildInstitutionAuthorizationContext(
+            $actorId,
+            new InstitutionRole(InstitutionRole::ROLE_USE_RA)
+        );
+
+        $paginator = $this->secondFactorService->searchVerifiedSecondFactors($query);
+
+        return JsonCollectionResponse::fromPaginator($paginator);
+    }
+
+    public function collectionWithoutAuthorizationContextAction(Request $request)
```

Comment:
I think this could be better renamed to be more descriptive for example `tokensOfIdentityCokkectionAction` or something like that. Because it can only be used by SS and is only used to list own tokens we should drop complexity if we can.

Reply:
Renamed to `collectionOfIdentityAction`